File: src/io/innofang/java/KMPTest.java
package io.innofang.java;

import io.innofang.java.string.KMP;

/**
 * Created by Inno Fang on 2018/3/15.
 */
public class KMPTest {
    public static void main(String[] args) {
        KMP kmp = new KMP("abcdabcy");
        boolean result = kmp.matches("abcxabcdabcdabcy");
        System.out.println(result);
    }
}

File: src/io/innofang/kotlin/KMPTest.kt
package io.innofang.kotlin

import io.innofang.java.string.KMP

/**
 * Created by Inno Fang on 2018/3/16.
 */

fun main(args: Array<String>) {
    val kmp = KMP("abcdabcy")

    val result = kmp.matches("abcxabcdabcdabcy")
    println(result)

    val index = kmp.find("abcxabcdabcdabcy")
    println(index)
}